python/nav/web/auth.py

```diff
@@ -48,6 +65,67 @@
 LOGIN_URL = '/index/login/'
 
 
+class RemoteUserConfigParser(NAVConfigParser):
+    DEFAULT_CONFIG_FILES = [join('webfront', 'webfront.conf')]
+    DEFAULT_CONFIG = u"""
+[remote-user]
+enabled=no
+"""
+_config = RemoteUserConfigParser()
+
+
+def get_login_url(request):
```

Comment:
As previously mentioned, like Uncle Bob, I prefer code to be ordered in a step-down manner - at least as long as the scoping rules of the underlying langauge supports it, and Python does.

The highest level of abstraction should come first (i.e. the important bits here), then the lower levels later, which is the natural order in which I would want to read them (i.e. I only need to scroll down to read the nitty-gritty details if I need them).

Reply:
My first real (non-assembler) language was Pascal. Pascal wouldn't compile if you had the big bits first and the components later. IIRC there was dedicated syntax to warn the compiler of an unexpected order of functions/procedures!